src/repositories/in-memory-db/inMemoryProjectRepository.ts

```diff
@@ -0,0 +1,47 @@
+import { randomUUID } from 'crypto'
+
+import { Prisma, Project } from '@prisma/client'
+import { ProjectRepository } from '../prisma/project-repository'
+
+export class InMemoryProjectRepository implements ProjectRepository {
+  public dbProject: Project[] = []
+
+  constructor() {
+    const project: Prisma.ProjectCreateWithoutUserInput = {
+      title: 'React',
+      description: 'Novo',
+      tags: 'nov, nn',
+      link: 'https:',
+    }
+
+    const userId = '1'
+
+    this.create(project, userId)
+  }
+
+  async findById(id: string) {
+    const project = this.dbProject.find((project) => project.id === id)
+
+    return !project ? null : project
+  }
+
+  async create(
+    data: Prisma.ProjectCreateWithoutUserInput,
```

Comment:
You don't need to receive that separately; 
Prisma has a type called UncheckedCreateInput where you will receive the user ID, and the types themselves won't concern whether an existing user exists or not.

You can read more about it [here ](https://www.prisma.io/docs/orm/prisma-client/type-safety#generated-uncheckedinput-types)

There is an example [here](https://github.com/MatheusSanchez/gym-check-in-api/blob/master/src/repositories/in-memory/in-memory-check-ins-repository.ts#L62) as well, which is a similar use case where the entities depend on each other."